Xertifyed-noob/SkillsDash
# bugfix/data-fetch-issues

## Summary
This pull request addresses various issues related to data fetching in the application.

## Changes
- Fixed CORS issue that prevented data from being fetched properly.
- Resolved state mutation issue occurring during data fetch operations.
- Fixed component re-rendering issue to ensure correct data display.

## Testing
- Ensure the application runs without errors after the fixes.
- Check the console logs in the browser to ensure data is rendered correctly and the state is maintained appropriately.
- Use Postman or cURL to send requests to the API endpoints and verify that data fetching works correctly.

## Additional Notes
- This fix resolves critical data fetch issues that were affecting the application's performance.
- Future improvements might include optimizing the data fetching process for better performance.
